src/bin.ts

```diff
@@ -7,13 +7,19 @@ import { hideBin } from 'yargs/helpers'
 import type { ArchiveFormat } from './archive.js'
 import { zipFunctions } from './main.js'
 
+declare global {
+  // eslint-disable-next-line no-var
+  var ZISI_CLI: boolean
+}
+
 // CLI entry point
 const runCli = async function () {
   // @ts-expect-error TODO: `destFolder` and  `srcFolder` are not being passed
   // back from `parseArgs()`.
   const { destFolder, srcFolder, ...options } = parseArgs()
 
   try {
+    global.ZISI_CLI = true
```

Comment:
does this need to be a global?

Reply:
Not necessarily, but the alternative would be to pass the flag through the complete function hierarchy. And as the binary of zisi seems it might be deprecated and removed at some point, this was by far the more straightforward solution.